appIcons.js

```diff
@@ -271,15 +271,18 @@ const MyAppIcon = new Lang.Class({
             case clickAction.MINIMIZE:
                 // In overview just activate the app, unless the acion is explicitely
                 // requested with a keyboard modifier
-                if (!Main.overview._shown || !modifiers){
-                    if (this.app == focusedApp)
+                if (!Main.overview._shown || modifiers){
```

Comment:
I think you meant `modifiers` and not `!modifiers`? I changed this to get the behaviour described in the comment just above.